src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/filesystem/BitbucketSCMFileSystem.java

```diff
@@ -86,7 +86,7 @@ public SCMFileSystem build(Item item, SCM scm, @CheckForNull SCMRevision scmRevi
 
             BitbucketFilePathClient filePathClient =
                     clientFactoryProvider.getClient(maybeServerConfiguration.get().getBaseUrl(),
-                            jenkinsToBitbucketCredentials.toBitbucketCredentials(repository.getCredentialsId()))
+                            jenkinsToBitbucketCredentials.toBitbucketCredentials(repository.getCredentialsId(), item))
```

Comment:
This is a good catch, but I don't think it relates to JENKINS-65541. This is ensuring our repository file endpoints have access to folder credentials when performing a lightweight checkout. A definite win, thanks for fixing :)